tests/main/writable-areas/task.yaml

```diff
@@ -14,11 +14,35 @@ environment:
 
 prepare: |
     snap pack data-writer
+    SNAP_MOUNT_DIR="$(os.paths snap-mount-dir)"
+    if [ "$SNAP_REEXEC" = "1" ] && [ "$SNAP_MOUNT_DIR" != "/snap" ] && [ ! -L /snap ]; then
+        # reexec expects to find the snapd snap under /snap, enable it to do so
```

Comment:
I wonder if this is a separate bug we should look into.

Reply:
I wouldn't say it's a bug, more like a thing we need to account for, but yeah it'd be nice to not needing the symlink, but the way snapd snap is currently built symlink is rather mandatory.